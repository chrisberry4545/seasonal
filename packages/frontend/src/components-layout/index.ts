export * from './ImageGridItem/ImageGridItem';

export * from './ImageGrid/ImageGrid';

export * from './SummaryColumn';

export * from './SummaryColumnList';

export * from './SeasonNameView/SeasonNameView';

export * from './Toast/Toast';
