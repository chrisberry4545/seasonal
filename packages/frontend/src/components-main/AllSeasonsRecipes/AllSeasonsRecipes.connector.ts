import { connect } from 'react-redux';
import {
  AllSeasonsRecipes
} from './AllSeasonsRecipes';

import { IState, selectIsCurrentTabRecipes } from '../../store';
import {
  IAllSeasonsRecipesInputProps,
  IAllSeasonsRecipesDispatchProps
} from './AllSeasonsRecipes.interface';
import { Dispatch } from 'redux';
import {
  selectAllSeasonsVisibleRecipesData,
  selectIsAllSeasonsRecipesLoading,
  increaseNumberOfAllSeasonsInView,
  selectHasMoreSeasonsInAllSeasonsView,
  recipeItemClicked
} from '@chrisb-dev/seasonal-shared-frontend-redux';

const mapStateToProps = (
  state: IState
): IAllSeasonsRecipesInputProps => ({
  hasMoreSeasonsInAllSeasonsView: selectHasMoreSeasonsInAllSeasonsView(state),
  isCurrentTabRecipes: selectIsCurrentTabRecipes(state),
  isLoading: selectIsAllSeasonsRecipesLoading(state),
  seasons: selectAllSeasonsVisibleRecipesData(state)
});

const mapDispatchToProps = (
  dispatch: Dispatch
): IAllSeasonsRecipesDispatchProps => ({
  increaseNumberOfAllSeasonsInView: () => (
    dispatch(increaseNumberOfAllSeasonsInView())
  ),
  onRecipeClick: (foodItemId: string) => dispatch(recipeItemClicked(foodItemId))
});

export const AllSeasonsRecipesConnector = connect(
  mapStateToProps,
  mapDispatchToProps
)(AllSeasonsRecipes);
