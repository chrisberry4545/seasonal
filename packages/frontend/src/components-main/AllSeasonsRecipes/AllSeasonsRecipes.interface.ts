import { IHydratedSeason } from '@chrisb-dev/seasonal-shared-models';

export interface IAllSeasonsRecipesInputProps {
  isCurrentTabRecipes: boolean;
  isLoading: boolean;
  seasons: IHydratedSeason[] | undefined;
  hasMoreSeasonsInAllSeasonsView: boolean;
}

export interface IAllSeasonsRecipesDispatchProps {
  increaseNumberOfAllSeasonsInView: () => void;
  onRecipeClick: (recipeId: string) => void;
}

export interface IAllSeasonsRecipesProps
  extends IAllSeasonsRecipesInputProps, IAllSeasonsRecipesDispatchProps {}
