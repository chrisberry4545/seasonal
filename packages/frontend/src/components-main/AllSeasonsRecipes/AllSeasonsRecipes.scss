@import '../../styles/settings';
@import '../../styles/tools';

.c-all-seasons-recipes__loading-spinner-wrapper {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.c-all-seasons-recipes {
  display: flex;
  flex-direction: column;
  width: 100%;
}

  .c-all-seasons-recipes__season {
    margin-bottom: 2rem;
    @include mq($from: tablet) {
      margin-bottom: 3rem;
    }
  }
