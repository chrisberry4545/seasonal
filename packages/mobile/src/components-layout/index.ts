export * from './ConfirmButtonLayout/ConfirmButtonLayout';

export * from './DefaultPaddingContainer/DefaultPaddingContainer';

export * from './DetailsPageLayout/DetailsPageLayout';

export * from './FeedbackQuestionsLayout/FeedbackQuestionsLayout';

export * from './HeaderAndBackButton/HeaderAndBackButton';

export * from './ImageGrid/ImageGrid';

export * from './ImageGridItem/ImageGridItem';

export * from './MainContainer/MainContainer';

export * from './ModalLayout/ModalLayout';

export * from './NoResultsText/NoResultsText';

export * from './SeasonNameView/SeasonNameView';

export * from './SwitchableGridOrList/SwitchableGridOrList';

export * from './Toast/Toast';

export * from './TopLoadingSpinner/TopLoadingSpinner';
