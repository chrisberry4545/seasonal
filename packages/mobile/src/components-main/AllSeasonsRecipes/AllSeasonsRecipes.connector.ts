import { connect } from 'react-redux';
import {
  AllSeasonsRecipes
} from './AllSeasonsRecipes';

import { IState } from '../../store';
import {
  IAllSeasonsRecipesInputProps,
  IAllSeasonsRecipesDispatchProps
} from './AllSeasonsRecipes.interface';
import { Dispatch } from 'redux';
import {
  selectIsAllSeasonsFoodLoading,
  selectAllSeasonsVisibleFoodData,
  increaseNumberOfAllSeasonsInView,
  toggleListView,
  selectIsListViewShown,
  recipeItemClicked
} from '@chrisb-dev/seasonal-shared-frontend-redux';

const mapStateToProps = (
  state: IState
): IAllSeasonsRecipesInputProps => ({
  isListViewShown: selectIsListViewShown(state),
  isLoading: selectIsAllSeasonsFoodLoading(state),
  seasons: selectAllSeasonsVisibleFoodData(state)
});

const mapDispatchToProps = (
  dispatch: Dispatch
): IAllSeasonsRecipesDispatchProps => ({
  increaseNumberOfAllSeasonsInView: () => (
    dispatch(increaseNumberOfAllSeasonsInView())
  ),
  onRecipeClick: (recipeId: string) => dispatch(recipeItemClicked(recipeId)),
  onToggleListView: () => dispatch(toggleListView())
});

export const AllSeasonsRecipesConnector = connect(
  mapStateToProps,
  mapDispatchToProps
)(AllSeasonsRecipes);
