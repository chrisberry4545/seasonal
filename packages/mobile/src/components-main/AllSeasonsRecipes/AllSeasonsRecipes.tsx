import React, { FC } from 'react';
import {
  IAllSeasonsRecipesProps
} from './AllSeasonsRecipes.interface';
import { AllSeasonsWrapper } from '../AllSeasonsWrapper/AllSeasonsWrapper';

export const AllSeasonsRecipes: FC<IAllSeasonsRecipesProps> = ({
  isLoading,
  increaseNumberOfAllSeasonsInView,
  onRecipeClick,
  seasons,
  isListViewShown,
  onToggleListView
}) => (
  <AllSeasonsWrapper
    propName='recipes'
    isLoading={isLoading}
    increaseNumberOfAllSeasonsInView={increaseNumberOfAllSeasonsInView}
    onItemClick={onRecipeClick}
    seasons={seasons}
    isListViewShown={isListViewShown}
    onToggleListView={onToggleListView} />
);
