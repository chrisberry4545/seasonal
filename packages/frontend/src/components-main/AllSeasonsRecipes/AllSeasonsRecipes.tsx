import React, { FC } from 'react';
import './AllSeasonsRecipes.scss';
import {
  IAllSeasonsRecipesProps
} from './AllSeasonsRecipes.interface';
import { SeasonNameView, ImageGrid } from '../../components-layout';
import { LoadingSpinner } from '@chrisb-dev/seasonal-shared-frontend-components';

import InfiniteScroll from 'react-infinite-scroller';
import { DietaryFiltersConnector } from '../DietaryFilters/DietaryFilters.connector';

export const AllSeasonsRecipes: FC<IAllSeasonsRecipesProps> = ({
  isCurrentTabRecipes,
  isLoading,
  onRecipeClick,
  seasons,
  hasMoreSeasonsInAllSeasonsView,
  increaseNumberOfAllSeasonsInView
}) => (
  isCurrentTabRecipes
    ? (
      isLoading
      ? <div className='c-all-seasons-recipes__loading-spinner-wrapper'>
          <LoadingSpinner />
        </div>
      : <div className='c-all-seasons-recipes'>
          <InfiniteScroll
            pageStart={0}
            loadMore={increaseNumberOfAllSeasonsInView}
            hasMore={hasMoreSeasonsInAllSeasonsView}
            loader={<LoadingSpinner key='all-seasons-spinner' />}>
          {
            seasons && seasons.map(({ name, recipes }) => (
              <div className='c-all-seasons-recipes__season' key={name}>
                <SeasonNameView name={name}></SeasonNameView>
                <DietaryFiltersConnector />
                <ImageGrid data={recipes}
                  onClick={onRecipeClick} skipAnimation={true} />
              </div>
            ))
          }
          </InfiniteScroll>
        </div>
    ) : null
);
