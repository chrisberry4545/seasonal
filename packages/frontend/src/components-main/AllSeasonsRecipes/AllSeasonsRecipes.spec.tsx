
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { AllSeasonsRecipes } from './AllSeasonsRecipes';
import { IAllSeasonsRecipesProps } from './AllSeasonsRecipes.interface';
import { IHydratedSeason } from '@chrisb-dev/seasonal-shared-models';
import { ImageGrid } from '../../components-layout';
import InfiniteScroll from 'react-infinite-scroller';
import { LoadingSpinner } from '@chrisb-dev/seasonal-shared-frontend-components';

jest.mock('../../components-layout', () => ({
  ImageGrid: () => 'ImageGrid',
  SeasonNameView: () => 'SeasonNameView'
}));
jest.mock('@chrisb-dev/seasonal-shared-frontend-components', () => ({
  LoadingSpinner: () => 'LoadingSpinner'
}));
jest.mock('../DietaryFilters/DietaryFilters.connector', () => ({
  DietaryFiltersConnector: () => 'DietaryFiltersConnector'
}));

describe('<AllSeasonsRecipes />', () => {
  let wrapper: ShallowWrapper;
  let mockOnRecipeClick: jest.Mock;
  const seasons = [{
    id: '1',
    name: 'January'
  }, {
    id: '2',
    name: 'February'
  }] as IHydratedSeason[];
  let mockIncreaseNumberOfAllSeasonsInView: jest.Mock;

  const initProps = (): IAllSeasonsRecipesProps => ({
    hasMoreSeasonsInAllSeasonsView: true,
    increaseNumberOfAllSeasonsInView:
      mockIncreaseNumberOfAllSeasonsInView,
    isCurrentTabRecipes: true,
    isLoading: false,
    onRecipeClick: mockOnRecipeClick,
    seasons
  });

  beforeEach(() => {
    mockOnRecipeClick = jest.fn();
    mockIncreaseNumberOfAllSeasonsInView = jest.fn();
  });

  describe('when the seasons are loaded', () => {
    beforeEach(() =>
      wrapper = shallow(
        <AllSeasonsRecipes {...initProps()} />
      )
    );

    test('renders correctly', () => expect(wrapper).toMatchSnapshot());

    test('when a food item is clicked', () => {
      const onClick = wrapper.find(ImageGrid).first().props().onClick;
      if (onClick) {
        onClick('id');
      }
      expect(mockOnRecipeClick).toHaveBeenCalledWith('id');
    });

    test('when more seasons are loaded', () => {
      const loadMore = wrapper.find(InfiniteScroll).first().props().loadMore;
      if (loadMore) {
        loadMore(1);
      }
      expect(mockIncreaseNumberOfAllSeasonsInView).toHaveBeenCalled();
    });

  });

  describe('when the seasons are not loaded', () => {
    beforeEach(() =>
      wrapper = shallow(
        <AllSeasonsRecipes {...({
          ...initProps(),
          isLoading: true
        })} />
      )
    );

    test('displays a loading spinner', () =>
      expect(wrapper.find(LoadingSpinner).exists()).toBe(true));

  });

  describe('when the current tab is not the recipes tab', () => {
    beforeEach(() =>
      wrapper = shallow(
        <AllSeasonsRecipes {...({
          ...initProps(),
          isCurrentTabRecipes: false
        })} />
      )
    );

    test('does not render anything', () =>
      expect(wrapper.children().exists()).toBe(false));

  });

});
