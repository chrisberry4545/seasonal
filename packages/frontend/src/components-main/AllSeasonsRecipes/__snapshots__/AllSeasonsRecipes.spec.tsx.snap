// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllSeasonsRecipes /> when the seasons are loaded renders correctly 1`] = `
<div
  className="c-all-seasons-recipes"
>
  <InfiniteScroll
    element="div"
    getScrollParent={null}
    hasMore={true}
    initialLoad={true}
    isReverse={false}
    loadMore={[MockFunction]}
    loader={<LoadingSpinner />}
    pageStart={0}
    ref={null}
    threshold={250}
    useCapture={false}
    useWindow={true}
  >
    <div
      className="c-all-seasons-recipes__season"
      key="January"
    >
      <SeasonNameView
        name="January"
      />
      <ImageGrid
        onClick={[MockFunction]}
        skipAnimation={true}
      />
    </div>
    <div
      className="c-all-seasons-recipes__season"
      key="February"
    >
      <SeasonNameView
        name="February"
      />
      <ImageGrid
        onClick={[MockFunction]}
        skipAnimation={true}
      />
    </div>
  </InfiniteScroll>
</div>
`;
